extensions/wikia/PortableInfobox/controllers/PortableInfoboxParserTagController.class.php

```diff
@@ -187,12 +193,32 @@ private function getThemeName( $params, $value ) {
 	}
 
 	private function getLayout( $params ) {
+		$validator = new InfoboxParamsValidator();
 		$layoutName = isset( $params[ 'layout' ] ) ? $params[ 'layout' ] : false;
-		if ( $layoutName && in_array( $layoutName, $this->supportedLayouts ) ) {
+		if ( $validator->validateLayout( $layoutName ) ) {
 			//make sure no whitespaces, prevents side effects
 			return self::INFOBOX_LAYOUT_PREFIX . $layoutName;
 		}
 
 		return self::INFOBOX_LAYOUT_PREFIX . self::DEFAULT_LAYOUT_NAME;
 	}
+
+	private function getColor( $colorName, $params, PPFrame $frame ) {
+		$validator = new InfoboxParamsValidator();
+
+		$sourceParam = $colorName . '-source';
```

Comment:
`colorName` is misleading, the first thing that came to my mind was "yellow" :wink: